OpenRA.Mods.Common/Lint/CheckTranslationReference.cs

```diff
@@ -16,19 +16,42 @@
 using System.Reflection;
 using Linguini.Syntax.Ast;
 using Linguini.Syntax.Parser;
+using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Lint
 {
-	class CheckTranslationReference : ILintPass
+	class CheckTranslationReference : ILintRulesPass
 	{
 		const BindingFlags Binding = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static;
 
 		readonly List<string> referencedKeys = new List<string>();
 		readonly Dictionary<string, string[]> referencedVariablesPerKey = new Dictionary<string, string[]>();
 		readonly List<string> variableReferences = new List<string>();
 
-		public void Run(Action<string> emitError, Action<string> emitWarning, ModData modData)
+		void ILintRulesPass.Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Ruleset rules)
 		{
+			// Custom maps may disable traits which triggers warnings about unused translations.
+			if (rules != modData.DefaultRules)
+				return;
```

Comment:
We should probably still lint when maps define redundant translation references in traits